okio/src/test/java/okio/BufferTest.java

```diff
@@ -163,6 +164,26 @@
     return segmentSizes;
   }
 
+  @Test public void largeByteStringReadWriteDoesNotCopy() throws Exception {
+    Buffer src = new Buffer();
+    List<Integer> segmentsSizes = new ArrayList<>();
+    for (int i=0, total=0; total <= Buffer.READBYTESTRING_SEGMENT_THRESHOLD; i++) {
+      int size = Segment.SIZE/2 + i;
+      src.writeAll(new Buffer().writeUtf8(repeat((char) ('a' + i), size)));
+      segmentsSizes.add(size);
+      total += size;
+    }
+    assertEquals(segmentsSizes, src.segmentSizes());
+
+    ByteString string = src.readByteString();
+    assertTrue("ByteString is segmented", string instanceof SegmentedByteString);
```

Comment:
The message is displayed when the assertion fails. In that case the statement "ByteString is segmented" is wrong. I suggest to either use "should be" or "expected…to be".

Same issue with the assertion further down.

Reply:
Thanks! Too used to AssertJ...